.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+# On every pull request, but only on push to master
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - '*'
+  pull_request:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    container: osgeo/gdal:ubuntu-full-${{ matrix.gdal-version }}
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
```

Comment:
we could add `3.10.0` to the list

Reply:
I would do just `'3.10'`, then patch versions would be installed. Don't forget the quotation marks, `3.10` will be interpreted as `3.1`.